skorch/__init__.py

```diff
@@ -3,15 +3,15 @@
 import sys
 import warnings
 
-import pkg_resources
-from pkg_resources import parse_version
+import importlib.metadata
```

Comment:
The tests are failing because `importlib.metadata` is only available `Python>=3.8`.

PyTorch has dropped Python 3.7 recently: https://github.com/pytorch/pytorch/commit/5976f0bdfed856ab48824bd41487318f3cf6e615 so I think it's okay for us to drop it as well.